### PR TITLE
LPS-124498 Include nested fields when checking the maximum repetitions

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://issues.liferay.com/browse/LPS-124498

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/checkbox/CheckboxDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -54,12 +62,42 @@
...
+				return getSystemSettingsURL(
+					ddmFormFieldRenderingContext.getHttpServletRequest());
+			}
 		).put(
 			"value",
 			GetterUtil.getBoolean(ddmFormFieldRenderingContext.getValue())
 		).build();
 	}
 
+	public String getSystemSettingsURL(HttpServletRequest httpServletRequest) {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/checkbox/CheckboxDDMFormFieldTemplateContextContributor.java:84]: @carolmariaabb, does it need to be `public` or can we make it `protected`/`private`? If it doesn't need to be `public`, can you please change it?

**Review by natocesarrego — COMMENTED** [review]
Terrific job, @carolmariaabb and @alinedoleron! Please see my unique comment. 🙂

**Comment by carolmariaabb** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/checkbox/CheckboxDDMFormFieldTemplateContextContributor.java (hunk 1 above)
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/checkbox/CheckboxDDMFormFieldTemplateContextContributor.java]: @natocesarrego I changed it to private. Thanks, I hadn't noticed it was public.

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**